internal/config/config.go

```diff
@@ -24,7 +24,7 @@ import (
 // clients can be specified either as an array or as a map. Some systems (e.g.,
 // Helm) do not have proper support for configuration values nested under
 // arrays, so just support both here.
-func ParseServicesConfig(raw json.RawMessage) (map[string]rest.Client, error) {
+func ParseServicesConfig(raw json.RawMessage, authPluginLookup func(string) rest.HTTPAuthPlugin) (map[string]rest.Client, error) {
```

Comment:
nit: this is a breaking change which is OK because the package is internal, however, I'd probably make this function take a struct now that we're going beyond a single parameter containing the raw config. This will insulate callers from similar changes in the future. Something like:

```golang
type ServiceOptions struct {
  Raw json.RawMessage
  AuthPlugin func(string) rest.HTTPAuthPlugin
}

func ParseServicesConfig(opts ServiceOptions) (map[string]rest.Client, error) { ... }
```